packages/wellknown-did-verifier/src/types/IWellKnownDidVerifier.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  IDidConfigurationResource,
+  IDomainLinkageValidation,
+  IVerifyCallbackArgs,
+  IVerifyCredentialResult,
+  IResourceValidation
+} from '@sphereon/wellknown-dids-client/dist/types'
+import {
+  IAgentContext,
+  IPluginMethodMap,
+  IResolver
+} from '@veramo/core'
+
+export interface IWellKnownDidVerifier extends IPluginMethodMap {
+  registerSignatureVerification(args: IRegisterSignatureVerificationArgs, context: IRequiredContext): Promise<void>,
+  removeSignatureVerification(args: IRemoveSignatureVerificationArgs, context: IRequiredContext): Promise<boolean>,
+  verifyDomainLinkage(args: IVerifyDomainLinkageArgs, context: IRequiredContext): Promise<IDomainLinkageValidation>,
+  verifyDidConfigurationResource(args: IVerifyDidConfigurationResourceArgs, context: IRequiredContext): Promise<IResourceValidation>
+}
+
+export interface IWellKnownDidVerifierOptionsArgs {
+  signatureVerifications?: Record<string, (args: IVerifyCallbackArgs) => Promise<IVerifyCredentialResult>>
+  onlyVerifyServiceDids?: boolean
+}
+
+export interface IRegisterSignatureVerificationArgs {
+  key: string
```

Comment:
Can we change the name. This might be confusing with did private/public keys

Reply:
renamed

Reply:
See other remarks. It can still be consufed with a public key for signature verification. Suggest to name it to something like callbackName, which makes immediate sense to a developer

Reply:
renamed